nipst/nipst.go

```diff
@@ -0,0 +1,380 @@
+package nipst
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/crypto"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"sync"
+	"time"
+)
+
+type postProof []byte
+
+func (p *postProof) Valid() bool {
+	return true
+}
+
+type PoetRound struct {
+	id uint64
+}
+
+type merkleProof []byte
+
+type membershipProof struct {
+	Root  common.Hash
+	Proof merkleProof
+}
+
+func (p *membershipProof) Valid() bool {
+	// TODO: implement
+	return true
+
+}
+
+// poetProof can convince a verifier that at least T time must have passed
+// from the time the initial challenge was learned.
+type poetProof []byte
+
+func (p *poetProof) Valid() bool {
+	// TODO: implement
+	return true
+}
+
+type SeqWorkTicks uint64
+
+type Space uint64
+
+// PostProver provides proving functionality for PoST.
+type PostProver interface {
```

Comment:
I think these APIs should be exported eventually when we integrate this into the actual node flow

Reply:
Since they are already public, do you mean to move them to a separate package?